libraries/classes/DbTableExists.php

```diff
@@ -4,112 +4,90 @@
         }
 
-        Core::sendHeaderLocation('./index.php?route=/' . Url::getCommonRaw($urlParams, '&'));
-
-        $response->callExit();
+        return false;
     }
 
-    private static function checkTable(string $db, string $table, bool $isTransformationWrapper): void
+    /**
+     * This "Commands out of sync" 2014 error may happen, for example after calling a MySQL procedure;
+     * at this point we can't select the db, but it's not necessarily wrong.
+     *
+     * @see https://dev.mysql.com/doc/mysql-errors/8.0/en/client-error-reference.html#error_cr_commands_out_of_sync
+     */
+    private function hasCommandsOutOfSyncError(): bool
     {
-        if (! empty($GLOBALS['is_table']) || defined('PMA_SUBMIT_MULT') || defined('TABLE_MAY_BE_ABSENT')) {
-            return;
-        }
-
-        $GLOBALS['is_table'] = false;
-        if ($table !== '') {
-            $GLOBALS['is_table'] = $GLOBALS['dbi']->getCache()->getCachedTableContent([$db, $table], false);
-            if ($GLOBALS['is_table']) {
-                return;
-            }
-
-            $result = $GLOBALS['dbi']->tryQuery('SHOW TABLES LIKE \'' . $GLOBALS['dbi']->escapeString($table) . '\';');
-            $GLOBALS['is_table'] = $result && $result->numRows();
-        }
-
-        if ($GLOBALS['is_table']) {
-            return;
-        }
+        return str_starts_with($this->dbi->getError(), '#2014 ');
+    }
 
-        if ($isTransformationWrapper) {
-            ResponseRenderer::getInstance()->callExit();
-        }
+    private function hasCachedTableContent(DatabaseName $database, TableName $table): bool
+    {
+        return (bool) $this->dbi->getCache()->getCachedTableContent([$database->getName(), $table->getName()]);
+    }
 
-        if ($table !== '') {
-            /**
-             * SHOW TABLES doesn't show temporary tables, so try select
-             * (as it can happen just in case temporary table, it should be fast):
-             */
-            $result = $GLOBALS['dbi']->tryQuery('SELECT COUNT(*) FROM ' . Util::backquote($table) . ';');
-            $GLOBALS['is_table'] = $result && $result->numRows();
-        }
+    private function isPermanentTable(TableName $table): bool
+    {
+        $result = $this->dbi->tryQuery(sprintf('SHOW TABLES LIKE %s;', $this->dbi->quoteString($table->getName())));
 
-        if ($GLOBALS['is_table']) {
-            return;
-        }
+        return $result !== false && $result->numRows() > 0;
+    }
 
-        /** @var SqlController $controller */
-        $controller = Core::getContainerBuilder()->get(SqlController::class);
-        $controller(Application::getRequest());
+    /**
+     * SHOW TABLES doesn't show temporary tables, so try select
+     * as it can happen just in case temporary table. It should be fast.
+     */
+    private function isTemporaryTable(TableName $table): bool
+    {
+        $result = $this->dbi->tryQuery(sprintf('SELECT COUNT(*) FROM %s;', Util::backquote($table)));
```

Comment:
```suggestion
        $result = $this->dbi->tryQuery(sprintf('SELECT 1 FROM %s LIMIT 1;', Util::backquote($table)));
```
I believe this should be faster.

Reply:
If this is faster, then it can also improve the MySQL `sys.table_exists` procedure.

https://github.com/mysql/mysql-sys/blob/a1374ea7cf341b71fd1060136533ce01fe0bd57b/procedures/table_exists.sql#L141

Reply:
No, it won't. They don't execute it. They only prepare.